src/Datadog.Trace/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,110 @@
+            if (pluginsConfig == null)
+            {
+                return;
+            }
+
+            var runtimeName = GetPluginsRuntimeName();
+
+            Log.Debug("Executing plugins configuration: {0}", pluginsConfig);
+            Log.Information("Trying to load plugins for '{0}' runtime.", runtimeName);
+
+            // TODO: Detect if objects instead of path
+            var pluginFiles = pluginsConfig?.GetValue<JToken>($"['{runtimeName}']").ToObject<string[]>();
+
+            if (pluginFiles == null || !pluginFiles.Any())
+            {
+                Log.Information("Skipping plugins load. Could not find any plugins for '{0}' runtime.", runtimeName);
+
+                return;
+            }
+
+            var loadedPlugins = TryLoadPlugins(pluginFiles);
+
+            Log.Information("Successfully loaded '{0}' plugin(s).", property: loadedPlugins.Count);
+
+            Loaded = loadedPlugins;
+        }
+
+        private static IReadOnlyCollection<IOTelPlugin> TryLoadPlugins(string[] pluginFiles)
+        {
+            var loaded = new List<IOTelPlugin>();
+
+            foreach (string file in pluginFiles)
+            {
+                string fullPath = Path.GetFullPath(file);
+
+                if (File.Exists(fullPath))
+                {
+                    try
+                    {
+                        Assembly pluginAssembly = Assembly.LoadFrom(fullPath);
+                        IOTelPlugin plugin = ConvertToPlugin(pluginAssembly);
+
+                        if (plugin != null)
+                        {
+                            loaded.Add(plugin);
+
+                            Log.Information("Plugin assembly loaded '{0}'.", pluginAssembly.FullName);
+                        }
+                        else
+                        {
+                            Log.Warning("Could not load {0} from '{1}'.", nameof(IOTelPlugin), pluginAssembly.FullName);
+                        }
+                    }
+                    catch (Exception ex) when (
+                        ExceptionUtil.IsAssemblyLoadException(ex) ||
+                        ExceptionUtil.IsDynamicInvocationException(ex))
+                    {
+                        Log.Warning(ex, "Plugin assembly could not be loaded.");
+                        Log.Information("Skipping vendor plugin load.");
+                    }
+                }
+                else
+                {
+                    Log.Warning("Plugin path is defined but could not find the path '{0}'.", fullPath);
+                }
+            }
+
+            return loaded;
+        }
+
+        private static IOTelPlugin ConvertToPlugin(Assembly assembly)
+        {
+            var pluginType = typeof(IOTelPlugin);
+
+            var pluginInstance = assembly
+                .GetTypes()
+                .Where(p => pluginType.IsAssignableFrom(p))
+                .Select(p => (IOTelPlugin)Activator.CreateInstance(p))
+                .FirstOrDefault();
```

Comment:
any reason why not keeping it as enumerable so someone can implement multiple `IOTelPlugin`s? the problem I see is that currently if someone implements the interface multiple times then we do not know which one would be loaded. it may be hard when troubleshooting as we even have nothing in logs

Reply:
Currently it is a hard requirement / limitation 1 plugin per assembly (see requirements). I don't know any good use case for multiple plugins in the same assembly. I think allowing less initially is better than allowing more that is not used mostly.

Reply:
I am OK with limiting to 1 plugin per assembly, but we should at least log if there were more (warning) and which one was really loaded.

Reply:
Added warning